Src/DependencyCollector/Net45.Tests/DependencyTrackingTelemetryModuleHttpTest.cs

```diff
@@ -345,6 +344,196 @@ public void OnBeginOnEndAreNotCalledForAppInsightsUrl()
+            }
+        }
+
+        [TestMethod]
+        [Timeout(500000)]
+        public void TestDependencyCollectorPostRequestsAreCollectedEventSource()
+        {
+            using (var module = new DependencyTrackingTelemetryModule())
+            {
+                module.DisableDiagnosticSourceInstrumentation = true;
+                module.ProfileQueryEndpoint = FakeProfileApiEndpoint;
+                module.Initialize(this.config);
+
+                var url = new Uri(LocalhostUrl);
+                HttpWebRequest request = WebRequest.CreateHttp(url);
+                request.Method = "POST";
+                request.ContentLength = 1;
+
+                using (new LocalServer(LocalhostUrl))
+                {
+                    using (var stream = request.GetRequestStream())
+                    {
+                        stream.Write(new byte[1], 0, 1);
+                    }
+
+                    using (request.GetResponse())
+                    {
+                    }
+                }
+
+                this.ValidateTelemetryForEventSource(this.sentTelemetry.Single(), url, request, true, "200");
+            }
+        }
+
+        [TestMethod]
+        [Timeout(5000)]
+        public void TestHttpRequestsWithQueryStringAreCollectedDiagnosticSource()
+        {
+            using (var module = new DependencyTrackingTelemetryModule())
+            {
+                module.ProfileQueryEndpoint = FakeProfileApiEndpoint;
+                module.Initialize(this.config);
+                Assert.IsTrue(DependencyTableStore.IsDesktopHttpDiagnosticSourceActivated);
+
+                var url = new Uri(LocalhostUrl + "123?q=123");
+                HttpWebRequest request = WebRequest.CreateHttp(url);
+
+                using (new LocalServer(LocalhostUrl))
+                {
+                    using (request.GetResponse())
+                    {
+                    }
+                }
+
+                this.ValidateTelemetryForDiagnosticSource(this.sentTelemetry.Single(), url, request, true, "200");
+            }
+        }
+
+        [TestMethod]
+        [Timeout(5000)]
+        public void TestHttpRequestsWithQueryStringAreCollectedEventSource()
+        {
+            using (var module = new DependencyTrackingTelemetryModule())
+            {
+                module.DisableDiagnosticSourceInstrumentation = true;
+                module.ProfileQueryEndpoint = FakeProfileApiEndpoint;
+                module.Initialize(this.config);
+
+                var url = new Uri(LocalhostUrl + "123?q=123");
+                HttpWebRequest request = WebRequest.CreateHttp(url);
+
+                using (new LocalServer(LocalhostUrl))
+                {
+                    using (request.GetResponse())
+                    {
+                    }
+                }
+
+                this.ValidateTelemetryForEventSource(this.sentTelemetry.Single(), url, request, true, "200");
```

Comment:
I mean that TestDependencyCollectorPostRequestsAreCollectedDiagnosticSource and TestDependencyCollectorPostRequestsAreCollectedEventSource have a lot in common.
TestHttpRequestsWithQueryStringAreCollectedDiagnosticSource and TestHttpRequestsWithQueryStringAreCollectedEventSource have a lot in common.
And TestDependencyCollectionDiagnosticSourceRedirect and TestDependencyCollectionEventSourceRedirect have a lot in common.
Basically there are three tests each duplicated once with the only difference being whether or not module.DisableDiagnosticSourceInstrumentation is set to true.  That setting could be parameterized and the Assert on IsDesktopHttpDiagnosticSourceActivated changed to match the parameter value and then you could have three "long" test helper functions that you call twice (once with true and once with false) rather than two copies of each of the three tests.